pocs/scheduler/constraint.py

```diff
@@ -159,3 +159,30 @@ def get_score(self, time, observer, observation, **kwargs):
 
     def __str__(self):
         return "Moon Avoidance"
+
+
+class AlreadyVisited(BaseConstraint):
+
+    """ Simple Already Visited Constraint
+
+    A simple already visited constraint that determines if the given `observation`
+    has already been visited before. If given `observation` has already been
+    visited then it will not be considered for a call to become the `current observation`.
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def get_score(self, time, observer, observation, **kwargs):
+        veto = False
+        score = self._score
+
+        observed_list = kwargs.get('observed_list', observer.observed_list)
+
+        for seq_time, previous_obs in observed_list.items():
```

Comment:
Actually, now that the OrderedDict is sorted out I think you can switch this back to `if observation.seq_time in observed_list` and avoid the loop.

Reply:
Awesome, didn't like the idea of the loop but I remembered that `seq_time` will be different for each `observation.